src/frontend/Pretty_printing.ml

```diff
@@ -47,32 +72,44 @@ let skip_comments loc =
   skipped :=
     !skipped
     @ List.filter_map (get_comments loc) ~f:(function
+        | `Include, l, loc ->
+            (* This prevents against bad behavior, but also really terrible but technically allowed things fail
+               For example, an if statement where the 'else' is entirely inside the include.
+               This makes the failure noisy rather than ever producing anything invalid for these. *)
+            Common.FatalError.fatal_error_msg
+              [%message
+                "Attempting to move include!"
+                  (l : string list)
+                  (loc : Middle.Location_span.t)]
```

Comment:
1. "Attempting to move include" is not the problem, it is a failed attempt at solving the problem. The error message should say something like "Cannot format #include in this position."
2. `FatalError` is meant for unreachable code, and advises the user to file a bug report. However, that is not appropriate here: tricky includes are allowed by the language and we don't plan to support pretty printing them. We need a new error mechanism for this case.
3. Another bad behavior is when the `if`branch is entirely inside the include but `else` is outside. That is not caught until the validation step at the very end of pretty printing.

Reply:
1. Agreed
2. I would like to encourage bug reports on this, at least for the first few versions, in case we miss any cases we _should_ support. I suppose its possible to say this outside of a `FatalError` however
3. Yeah, that's a bit tricky. Because we don't keep track of the individual locations of `if` and `else`, it's impossible to tell if one of them was included from or not at the moment.

Reply:
I've been trying for a little bit and it's actually rather difficult to hit this case. Most of the 'bad' inputs I have formulated end up failing inside check_correctness, which is a `FatalError` anyway.

Personally, I'm okay with getting a few false positive issues opened about this. If anything it will help us calibrate how much utility there would be to users if we tried really hard to improve this feature past where it is now